docs/DevGuide/Observers.md

```diff
@@ -0,0 +1,119 @@
+the reductions on a node are complete, the final reduction is done to node
+`0`. This is an arbitrary choice, but we are always guaranteed to have a node
+`0`.
+
+Once all the reductions are received on node `0`, the `ObserverWriter` invokes
+the `InvokeFinal` (third) template parameter on each `Parallel::ReductionDatum`
+(this is the n-ary) in order to finalize the data before writing. This is used,
+for example, for dividing by the total number of grid points in an L1 or L2
+norm. The reduction data is then written to an HDF5 file whose name is set in
+the input file using the option
+`observers::Tags::ReductionFileName`. Specifically, the data is written into an
+`h5::Dat` subfile since, along with the data, the subfile name must be passed
+through the reductions.
+
+The actions used for registering reductions are
+`observers::Actions::RegisterEventsWithObservers`,
+`observers::Actions::RegisterSingletonWithObserverWriter`, and
+`observers::Actions::RegisterWithObservers`. There is a separate `Registration`
+phase at the beginning of all simulations where everything must register with
+the observers. The action `observers::Actions::ContributeReductionData` is used
+to send data to the `observers::Observer` component in the case where there is a
+reduction done across an array or subset of an array. If a singleton parallel
+component needs to write data directly to disk it should use the
+`observers::ThreadedActions::WriteReductionData` action called on the zeroth
+element of the `observers::ObserverWriter` component.
+
+### Volume Data
+
+Volume data is vaguely defined as anything that has some extent. For example, in
+a 3d simulation, data on 2d surfaces is still considered volume data for the
+purposes of observing data. The spectral coefficients can also be written as
+volume data, though some care must be taken in that case to correctly identify
+which mode is associated with which terms in the basis function
+expansion. Whatever component will contributed volume data to be written must
+register with the `observers::Observer` component (there currently isn't tested
+support for directly registering with the `observers::ObserverWriter`). This
+registration is the same as in the reduction data case.
+
+Once the observers are registered, data is contributed to the
+`observers::Observer` component using the
+`observers::Actions::ContributeVolumeData` action. The data is packed into a
+`std::vector<TensorComponent>`, where the `TensorComponent` is data
+from just one tensor component or a reduction over a tensor. The `extents`,
+`Spectral::Basis` and `Spectral::Quadrature` are currently also passed to the
+`ContributeVolumeData` action. Once all the elements on a single core have
+contributed their volume data to the `observers::Observer` group, the
+`observers::Observer` group moves its data to the `observers::ObserverWriter`
+component to be written. We write one file per node append the node ID to the
```

Comment:
The sentence 'We write one file per node append the node ID to the HDF5 file name' appears malformed, but I'm unsure what was intended.